vertx-sql-client/src/main/java/io/vertx/sqlclient/impl/SocketConnectionBase.java

```diff
@@ -161,6 +161,14 @@ private void checkPending() {
           paused = true;
           inflight++;
           cmd = prepareCmd;
+        } else {
+          // cached prepared statement needs to prepare the parameters before execution
+          String msg = queryCmd.prepare();
+          if (msg != null) {
```

Comment:
can we add a general test for this ?

Reply:
is this related to the actual bug reported ?

Reply:
we need this which was overlooked before, because the parameters of new execution need to be prepared for cached one-shot prepared queries otherwise the parameters would be prepared only once for the first time and never be prepared later.